src/init.cpp

```diff
@@ -1009,7 +1009,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         if (!warningString.empty())
             InitWarning(warningString);
         if (!errorString.empty())
-            return InitError(warningString);
+            return InitError(errorString);
```

Comment:
Good catch!
This should go in definitively.
Splitting the PR into chunks that make sense and are independent would be nice.